src/compiler/utilities.ts

```diff
@@ -1913,6 +1914,12 @@ namespace ts {
         return token >= SyntaxKind.FirstAssignment && token <= SyntaxKind.LastAssignment;
     }
 
+    export function isClassExtendsExpressionWithTypeArguments(node: Node): boolean {

```

Comment:
I need the function to take type `Node` and check the kind. Can't think of a better name.